### DISCUSSION TITLE
ethers v6

### DISCUSSION THREAD

**Discussion by songkeys** [wevm/wagmi#1782] [Ideas]
`ethers` just release v6. [Migration guides here](https://docs.ethers.org/v6/migrating/). Do we have a plan to migrate to it? I could help make a PR :)

Edit: I am doing the PR but keep finding some [issues](https://github.com/ethers-io/ethers.js/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Av6). I'd suggest we say "install `ethers@5`" in wagmi doc to avoid confusions for now. Meanwhile, I'll keep following these issues.

**Comment by tmm** [comment]
Since it was just released, let's hold off on the PR for now until we see how v6 progresses and issues are resolved.

**Comment by lekevicius** [comment]
Curious if there's anything specific on ethers 6 that would trigger the migration effort, or just time?

**Reply by jxom** [reply]
We will share more updates on our plans for this within the next fortnight.

**Reply by challet** [reply]
[Errors subclasses](https://docs.ethers.org/v6-beta/api/utils/errors/) look like the better improvement to me.

**Comment by lekevicius** [comment]
@jxom Just pinging to communicate here as well.

As I understand, wagmi will never migrate to ethers v6, and instead will be based on / migrate to [viem](https://viem.sh/)?

**Reply by tmm** [reply]
Correct, we feel that using viem will be a significant improvement for users across performance, bundle size, type-safety, etc. Also, since the wagmi core team is working on viem and wagmi, the integration between the two libraries will be solid.

**Reply by oveddan** [reply]
nice, I cannot wait to be able to switch to viem from ethers.js - how far off is that possibility to use viem with wagmi?

**Comment by hammeiam** [comment]
@tmm is there an estimate for how far away a `viem`-supporting version of wagmi will be?

**Reply by tmm** [reply]
Soon! Stayed tuned.

https://twitter.com/wagmi_sh/status/1643370731030585344

**Comment by jxom** [comment]
See more in this discussion: [The Future of wagmi](https://github.com/wagmi-dev/wagmi/discussions/2207)
